src/codec/meta/actor_settings.rs

```diff
@@ -1,22 +1,20 @@
 use ecdsa::signature::rand_core::CryptoRngCore;
 use futures::{AsyncWrite, AsyncWriteExt};
-use winnow::binary::le_u8;
-use winnow::token::take;
-use winnow::Parser;
+use winnow::{binary::le_u8, token::take, Parser};
 
-use crate::codec::crypto::{
-    AccessKey, AsymLockedAccessKey, AsymLockedAccessKeyError, SigningKey, VerifyingKey,
+use crate::codec::{
+    crypto::{AccessKey, AsymLockedAccessKey, AsymLockedAccessKeyError, SigningKey, VerifyingKey},
+    header::AccessMask,
+    meta::{UserAgent, VectorClockActorSnapshot},
+    ParserResult, Stream,
 };
-use crate::codec::header::AccessMask;
-use crate::codec::meta::{UserAgent, VectorClock, VectorClockSnapshot};
-use crate::codec::{ParserResult, Stream};
 
 const KEY_PRESENT_BIT: u8 = 0b0000_0001;
 
 #[derive(Clone, Debug)]
 pub struct ActorSettings {
     verifying_key: VerifyingKey,
-    vector_clock: VectorClock,
+    vector_clock: VectorClockActorSnapshot,
```

Comment:
It is redundant, but I purposely didn't export the InnerClock type, leaving it as an internal implementation detail.  But yeah it is definitely redundant.  I think generally we could, either lean into the redundancy and use it as an opportunity to do `strict` checks, ignore the redundancy, or rework the type structure of the Clock types. It looks like of the current use-cases of VectorClockActor, ActorId is redundant in all or almost all of them? Maybe The ActorId doesn't need to be stored alongside it by default? The only place it absolutely has to be stored alongside it will be the journal entries right?

Reply:
I think this is good, its a small thing and make sense in the current context 👍🏼